### PR TITLE
Add an endpoint for reaction summaries

### PR DESCRIPTION
**Pull request by skearnes**
Adds an API endpoint that takes a serialized Reaction message (generated by `reaction.serializeBinary()`) and returns a JSON dict of dicts containing reaction summary information. There are two sets of information:
* `provenance`: Dictionary of reaction metadata that should be shown immediately under the reaction ID.
* `summary`: Dictionary of conditions, analysis, notes, etc. info that should be shown under the schematic.

### CODE REVIEW

**Review by daniil-sloboda — COMMENTED** [review]
LGTM, up to @lopatinay whether to merge as is

File: ord_app/api/editor/reactions.py [hunk 1]
```diff
@@ -94,3 +95,12 @@ def delete_reaction(user_id: str, dataset_name: str, index: int):
...
         add_dataset(user_id, dataset, cursor)
     return Response(status_code=200)
+
+
+@router.post("/summarize_reaction")
+async def summarize_reaction(request: Request):
+    """Returns summary information for the reaction card view."""
+    # TODO(skearnes): Implement something useful here.
+    reaction = Reaction.FromString(await request.body())
+    del reaction  # Unused.
```
Comment (lopatinay) [inline ord_app/api/editor/reactions.py:105]: Minor: it's not necessary, Python's GC will delete this object on its own.

**Review by lopatinay — APPROVED** [review]
LGTM, left minor comment

File: ord_app/api/editor/reactions.py (hunk 1 above)
Comment (skearnes) [reply ord_app/api/editor/reactions.py:105]: This is about lint; see "Unused" here: https://google.github.io/styleguide/pyguide.html#214-decision

**Pull request merged**